src/main/java/tc/oc/pgm/rotation/MapPoll.java

```diff
@@ -222,7 +218,16 @@ private int countVotes(Set<UUID> uuids) {
    * @return The picked map to play after the vote
    */
   MapInfo finishVote() {
-    MapInfo picked = getMostVotedMap();
+    // Cache vote totals, so we don't have to count for each compare
+    for (Entry<MapInfo, Set<UUID>> map : votes.entrySet()) {
+      finalTotals.put(map.getKey(), countVotes(map.getValue()));
+    }
+    // Populate final order
+    finalTotals.entrySet().stream()
+        .sorted(Comparator.comparingInt((e) -> ((Entry<MapInfo, Integer>) e).getValue()).reversed())
+        .forEachOrdered(e -> finalOrder.add(e.getKey()));
```

Comment:
I'd rather not change the order of maps
Also, if you're using a stream, don't use a terminal operation that mutates a pre-made collection, just collect it to a list and assign it